casper/src/test/scala/coop/rchain/casper/util/BlockHashSignatureSpec.scala

```diff
@@ -2,25 +2,74 @@ package coop.rchain.casper.util
 
 import com.google.protobuf.ByteString
 import coop.rchain.casper.protocol.{BlockMessage, Justification}
-import org.scalatest.prop.PropertyChecks
-import org.scalatest.{FlatSpec, Matchers}
 import coop.rchain.models.blockImplicits._
+import org.scalacheck.Prop.{forAll, AnyOperators}
+import org.scalatest.FlatSpec
+import org.scalatest.prop.Checkers.check
 
-class BlockHashSignatureSpec extends FlatSpec with Matchers with PropertyChecks {
+class BlockHashSignatureSpec extends FlatSpec {
 
-  implicit val aBlock     = arbBlockMessage
-  implicit val aValidator = arbitraryValidator
+  implicit val aBlock = arbBlockMessage
 
-  "block hash" should "be created from all fields except signature" in {
-    forAll { (block: BlockMessage, bytes: ByteString) =>
-      val hash = ProtoUtil.hashBlock(block)
-      val blockModifiedJustifications = block.copy(
-        justifications = List(Justification(bytes, bytes))
-      )
-      val hashModified = ProtoUtil.hashBlock(blockModifiedJustifications)
+  "block hash" should "not match for changed block hash" in {
+    implicit val aBlockHash = arbitraryBlockHash
+    check {
+      forAll { (block: BlockMessage, genBlockHash: ByteString) =>
+        val isModified    = genBlockHash != block.blockHash
+        val hash          = block.blockHash
+        val blockModified = block.copy(blockHash = genBlockHash)
+        val hashModified  = ProtoUtil.hashBlock(blockModified)
 
-      hashModified shouldBe hash
+        // Check hashes are different for modified block
+        hashModified !== hash ?= isModified
+      }
     }
   }
 
+  "block hash" should "not match for changed sequence number" in {
```

Comment:
Do we need more of these for all fields in block?

Reply:
I'd like to have it. It's basically the same pattern for each field so good candidate for some macro to simplify.